[조이Zoey] 메인 페이지 주요 기능 구현

youzysu opened the pull request:
https://user-images.githubusercontent.com/111998760/227733544-ad0dc2ad-81ba-4897-8dd3-e23002ef1a48.mov

### ✏️ [구현 기능 체크리스트 및 세부 내용 링크](https://www.notion.so/60029378aeb64c029cdfe27498de6d59)
### 메인 페이지 주요 기능 구현
- 로컬 스토리지 동기화
- 상단의 수입지출 체크박스를 통해 수입, 지출 내역 필터링하여 렌더링하기
- 삭제하기 버튼 클릭 시 alert창 띄우기
  - 삭제 버튼 클릭 시 해당 내역 삭제
  - 취소 버튼 클릭 시 창닫기
- Modal 컴포넌트 구현
- 내역 수정하기 기능

youzysu committed:
refactor: 로컬 스토리지 저장을 위해 Map을 일반 객체로 변경, private field 삭제

- Map 객체는 localStorage에서 지원하지 않는 형식이고, private 프로퍼티도 저장되지 않는다.
- 데이터 저장을 위해 일반 객체로 수정하고, private field 삭제

youzysu committed:
refactor: 로컬스토리지에 저장되어있는 데이터 init 해오도록 리팩터링

Kakamotobi commented on src/js/HistoryStorage/HistoryStorage.js:
```diff
@@ -1,33 +1,72 @@
 import { MonthlyHistory } from './MonthlyHistory.js';
 
-export class HistoryStorage {
+class HistoryStorage {
```
"History"라는 이름이 모호한거 같아요. "Entry"같은 다른 단어를 사용하시거나 "History" 앞에 적절한 단어를 포함해주시면 좋을거 같아요.

youzysu replied:
말씀처럼 기획서에 내역들을 History라고 표현해서 그대로 사용했는데, Storage의 history를 의미하는걸로 오해할 수도 있겠네요. 클래스명 고민해보겠습니다.

Kakamotobi commented on src/js/InputBar/event/formSubmitHandler.js:
```diff
@@ -23,9 +23,14 @@ export const formSubmitHandler = (e) => {
-  changeHeaderMonthYear({ year, monthNumber, monthChar });
+    const { year, monthNumber, monthChar } = getDateFormat(dailyItemData.date);
+    changeHeaderMonthYear({ year, monthNumber, monthChar });
+  } catch (e) {
```
event랑 error 줄임을 구분해주시면 좋을것 같아요. (Ex: event - `evt`, error - `err`).

youzysu replied:
넵, 명확하게 사용하겠습니다. 좋은 의견 감사합니다!

gunoc commented on src/js/HistoryStorage/DailyHistory.js:
```diff
@@ -1,38 +1,10 @@
-    this.#fullDate = fullDate;
-    this.dailyItems = new Map();
+    this.fullDate = fullDate;
+    this.dailyItems = {};
```
이렇게도 데이터를 정리할 수 있구나 알 수 있었고 원하는 방식대로 조작하기 위해 다양하게 시도하고 계신 점에서 많이 배워갑니다!👍 하지만 데이터 한 줄에 객체 안 객체, 그안에 다시 객체 방식으로 저장되는 구조가 처음 읽었을 때 조금 복잡한 것 같다는 느낌을 받았습니다

youzysu replied:
데이터 구조를 포함 관계로 생각해서 이렇게 관리했는데, 저도 복잡해져서 오히려 구현할 때 불편한 부분도 있었네요. 구현하기 전에 설계할 때 발생할 수 있는 문제에 대해 더 고민해보는 시간을 가져야겠어요. 감사해요 :)

Kakamotobi reviewed (commented):
- dailyHistoryView.js line 45:
  - `classList`는 read-only로서 `classList = 'daily-history__dateChar'`보다 `classList.add('daily-history__dateChar')`이 더 적절한거 같아요.

- Barrel export에 대해 알아보셔도 좋을거 같아요. 같은 곳에서 너무 많은 것들을 import할 경우에 유용할 것 같아요.

- View랑 EventHandler들이 그렇게 길지 않은데 하나의 파일로 통합해도 괜찮을 것 같아요. 개인적으로는 작업하기도 훨씬 편한 것 같아요.

- 파일/폴더 네이밍 때문에 전체적인 구조가 조금 헷갈리는거 같아요. 예를 들어서, HistoryStorage라고 하지 않고 js폴더에서 빼거나 이름을 좀 더 general하게 (Ex: storage, database)바꿔주시면 더 빠르게 파악이 될것 같아요.

youzysu committed:
fix: 일자별 내역 객체 만드는 함수에 필요한 인자 추가하기

youzysu committed:
refactor: Date 관련 로직, 수입지출 데이터 처리 로직 전반 리팩터링

- Storage 객체에 수입지출 데이터 저장 및 관리 전반 수정
- Date 관련 로직 전반 수정: 인자로 Date 객체 자체를 넘겨서 내부적으로 필요한 형식으로 가공
- 아이템 저장 시 해당 월의 전체 개수, 수입 지출 총액 계산하도록 수정

youzysu committed:
refactor: Date 관련 로직, 수입지출 데이터 처리 로직 전반 리팩터링

- Storage 객체에 수입지출 데이터 저장 및 관리 전반 수정
- Date 관련 로직 전반 수정: 인자로 Date 객체 자체를 넘겨서 내부적으로 필요한 형식으로 가공
- 아이템 저장 시 해당 월의 전체 개수, 수입 지출 총액 계산하도록 수정

youzysu committed:
Merge branch 'feature/mainpage' of ssh://github.com/youzysu/fe-max--wise-wallet into feature/mainpage

youzysu committed:
refactor: 디렉터리 구조 리팩터링

youzysu committed:
fix: classList를 className으로 수정

- classList는 DomTokenList로 하나의 클래스 조작 시 사용
- 전체 클래스명을 initialize 해주기 위해 className으로 수정
- Initialize class attribute of element using string value

youzysu committed:
feat: 표시된 수입액과 지출액 앞에 배치한 체크박스 여부에 따라 화면에 표시한다.

- 리팩터링 필요

youzysu committed:
fix: 수입액, 지출액 체크여부에 따른 내역 렌더링 리팩터링

- 체크되지 않은 해당 element node가 아닌 빈문자열을 반환하여 DOM에 추가되지 않도록 수정
- 조건에 부합하는 경우 해당 노드 바로 리턴하도록 수정

youzysu committed:
refactor: 가독성을 위한 코드 전반 리팩터링

youzysu committed:
feat: [삭제하기 버튼] 클릭 시 해당 수입지출 내역 삭제 모달창을 띄우기

youzysu committed:
feat: 모달창 띄울 때 배경 딤처리하기

youzysu committed:
feat: 삭제 버튼 클릭 시 해당 내역을 삭제하고, 취소 버튼 클릭 시 alert 창만 닫는다.

youzysu committed:
feat: 수입지출 내역을 수정을 위한 inputBar 기능 구현

- 수입 지출 내역을 클릭하면 해당 내역의 내용을 입력바에 입력한다.
- 기존 내역과 수정된 내역을 비교하여 다른 경우 확인 버튼을 활성화한다.
